include/shark/Algorithms/Trainers/CSvmTrainer.h

```diff
@@ -251,6 +266,9 @@ class CSvmTrainer : public AbstractSvmTrainer<
 			typedef SvmShrinkingProblem<SVMProblemType> ProblemType;
 			ProblemType problem(svmProblem,base_type::m_shrinking);
 			QpSolver< ProblemType > solver(problem);
+			if (svm.alpha().size1() > 0 && svm.alpha().size2() == 1) {
+				problem.setInitialSolution(blas::column(svm.alpha(), 0));
```

Comment:
i think this will assert if you change C between two runs and the new C is smaller than the old.
also note that the if-statement is always true (setStructure in line 165 will create an alpha matrix with proper size).